services/ui/ws/display.cc

```diff
@@ -369,9 +373,15 @@ void Display::OnWindowStateChanged(ui::mojom::ShowState new_state) {
   if (!window_server_->IsInExternalWindowMode())
     return;
 
+  if (root_->GetShowState() == new_state)
```

Comment:
It would be better to have this check on PlatformDisplayDefault::OnWindowStateChanged level.

if (delegate_->GetRootWindow()->GetShowState() == state)
    return;